cloudinit/sources/DataSourceEc2.py

```diff
@@ -731,28 +735,91 @@ def convert_ec2_metadata_network_config(network_md, macs_to_nics=None,
        not provided, get_interfaces_by_mac is called to get it from the OS.
     @param: fallback_nic: Optionally provide the primary nic interface name.
        This nic will be guaranteed to minimally have a dhcp4 configuration.
+    @param: apply_network_config: Boolean set True to configure all networking
+       presented by IMDS. This includes rendering secondary IPv4 or IPv6
+       addresses on any NICs and rendering network config on secondary NICs.
```

Comment:
address on all NICs.  If False, only the primary nic will be configured and only with dhcp (IPv4/IPv6).

Reply:
thanks added.